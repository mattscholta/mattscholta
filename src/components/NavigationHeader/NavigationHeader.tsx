import * as React from 'react';
import { color } from '@barguide/style-guide';
import classnames from 'classnames';
import Link from 'next/link';

import { Logo } from '@src/components/Logo';
import { useRoutes } from '@src/hooks/useRoutes';

import styles from './styles.module.css';

interface NavigationHeaderProps {
  className?: string;
}

/**
 * @name NavigationHeader
 * @description tbd...
 */
const NavigationHeader: React.FC<NavigationHeaderProps> = (props) => {
  const { className } = props;

  // Hooks
  const { active } = useRoutes();

  // Styles
  const tailwind = `ui-container-xl u-flex u-items-center`;
  const cssActive = styles.active;
  const cssComponent = classnames('ui-shadow', styles.component, className);
  const cssLinks = classnames(tailwind, styles.links);

  const cssAbout = classnames(styles.link, { [cssActive]: active.homepage });
  const cssPhotos = classnames(styles.link, { [cssActive]: active.photos });
  const cssContact = classnames(styles.link, { [cssActive]: active.contact });

  return (
    <div className={cssComponent}>
      <nav className={cssLinks}>
        {/*
        <Link href="/">
          <Logo className={styles.logo} fill={color.white} />
        </Link>
        */}
        <Link href="/">
          <a className={cssAbout}>About</a>
        </Link>
        <Link href="/photos">
          <a className={cssPhotos}>Portfolio</a>
        </Link>
        <Link href="/contact">
          <a className={cssContact}>Ramblings</a>
        </Link>
        <Link href="/uses">
          <a className={cssContact}>Uses</a>
        </Link>
      </nav>
    </div>
  );
};

export { NavigationHeader };
export type { NavigationHeaderProps };
