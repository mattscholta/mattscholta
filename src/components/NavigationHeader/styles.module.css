.component {
  background-color: theme('colors.secondary');
  border: none;
  color: theme('colors.white');
  height: var(--header-height);
  position: sticky;
  top: 0;
  z-index: 9;
}

.links {
  height: 100%;
  overflow: hidden;
  padding: 0 theme('spacing.2x');
}

.link {
  background-color: transparent;
  color: theme('colors.white');
  padding: theme('spacing.2x');
  text-decoration: none;
  transition: background 140ms ease-out, color 140ms ease-out;
}

.link:active,
.link:hover {
  background-color: theme('colors.primary');
  color: theme('colors.secondary');
}

.link.active {
  background-color: theme('colors.primary');
  color: theme('colors.secondary');
  cursor: default;
  pointer-events: none;
}

.logo {
  padding: theme('spacing.2xs') 0;
}
