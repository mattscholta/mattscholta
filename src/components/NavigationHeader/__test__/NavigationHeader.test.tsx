import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { NavigationHeader, NavigationHeaderProps } from '../NavigationHeader';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('NavigationHeader', () => {
  let props: NavigationHeaderProps;
  let wrapper: RenderResult;

  const createWrapper = () => <NavigationHeader {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
