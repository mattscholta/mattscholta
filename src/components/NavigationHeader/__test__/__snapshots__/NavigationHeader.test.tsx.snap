// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationHeader matches snapshot 1`] = `
NodeList [
  <div
    class="ui-shadow component"
  >
    <nav
      class="ui-container-xl u-flex u-items-center links"
    >
      <svg
        class="logo"
        height="59"
        viewBox="0 0 100 59"
        width="100"
      >
        <title>
          mattscholta
        </title>
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <path
            d="M72.573561,53.8100014 L67.7974434,43 L63.0213258,53.8100014 L57.7974434,53.8100014 L52.573561,53.8100014 L47.7974434,43 L43.0213258,53.8100014 L24.0228956,53.8100014 L47.7974434,0 L57.7974434,22.6334489 L67.7974434,0 L91.5719913,53.8100014 L72.573561,53.8100014 Z M7.10754499,21.7630003 L16.7229599, 8.8817842e-16 L19.5285978,0 L8.51036391,24.9380633 L7.10754499, 21.7630003 Z M9.82164089,28.3034712 L22.3267926,2.0539126e-15 L26.1647839,0 L11.7406366,32.6468202 L9.82164089,28.3034712 Z M28.7229599,0 L44.5719913,0 L20.7974434,53.8100014 L16.7974434, 44.7566218 L36.5719913, 0 L28.7229599,0 L28.7229599,8.8817842e-16 Z M28.7229599,8.8817842e-16 L28.7229599, 8.8817842e-16 L34.1647839, 0 L15.6106287,41.9944523 L12.8897168,35.83609 Z M5.35664415, 14.1239332 L0,2 L10.7132883,2 L5.35664415,14.1239332 Z"
            fill="#fff"
          />
        </g>
      </svg>
      <a
        class="link"
        href="/"
      >
        About
      </a>
      <a
        class="link"
        href="/photos"
      >
        Portfolio
      </a>
      <a
        class="link"
        href="/contact"
      >
        Ramblings
      </a>
      <a
        class="link"
        href="/uses"
      >
        Uses
      </a>
    </nav>
  </div>,
]
`;
